openedx/core/djangoapps/external_user_ids/models.py

```diff
@@ -103,3 +103,50 @@ def add_new_user_id(cls, user, type_name):
                 )
             )
         return external_id, created
+
+    @classmethod
+    def batch_get_or_create_user_ids(cls, users, type_name):
+        """
+        Create ExternalIds in batch.
```

Comment:
Can you please provide an example of when it's useful to create these in a batch, instead of creating them for each user as needed?

Reply:
@bradenmacdonald, In the case of LTI Consumer, External ID's are created when a student actually launches LTI from LMS. But when providing course member information to a third party tool, not every member has External ID's available. To create one by one would be a performance issue. Thus this method has been created.

Reply:
@shimulch Oops, I meant to specify that I'd like that added to the docstring. Dosctrings should generally say both _what_ the code does and _why_ it exists, because without the why it's hard to know when/how it can be refactored or removed in the future.